src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -175,11 +176,233 @@ private BoundExpression BindSwitchExpression(SwitchExpressionSyntax node, Bindin
+                          PerformPatternIndexerLookup(syntax, receiverType, lookupResult, out indexerAccess, Compilation.GetSpecialType(SpecialType.System_Int32));
+            lookupResult.Free();
+            return success;
+        }
+
+        private bool TryFindRangeIndexerPatternForListPattern(
+            SyntaxNode syntax, TypeSymbol receiverType,
+            out MethodSymbol? sliceMethod,
+            out BoundIndexerAccess? indexerAccess)
+        {
+            sliceMethod = null;
+
+            var useSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
+            var lookupResult = LookupResult.GetInstance();
+            var success = PerformPatternIndexerLookup(syntax, receiverType, lookupResult, out indexerAccess, Compilation.GetWellKnownType(WellKnownType.System_Range)) ||
+                          TryFindImplicitRangeIndexerPattern(syntax, receiverOpt: null, receiverType, out sliceMethod, BindingDiagnosticBag.Discarded, lookupResult, ref useSiteInfo);
+            lookupResult.Free();
+            return success;
+        }
+
+        private bool PerformPatternIndexerLookup(SyntaxNode syntax, TypeSymbol receiverType, LookupResult lookupResult, out BoundIndexerAccess? valid, TypeSymbol? indexerArgumentType)
+        {
+            var diagnostics = BindingDiagnosticBag.Discarded;
+            var useSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
+
+            if (indexerArgumentType is not null)
+            {
+                LookupMembersInType(
+                    lookupResult,
+                    receiverType,
+                    WellKnownMemberNames.Indexer,
+                    arity: 0,
+                    basesBeingResolved: null,
+                    LookupOptions.Default,
+                    originalBinder: this,
+                    diagnose: false,
+                    ref useSiteInfo);
+
+                if (lookupResult.IsMultiViable)
+                {
+                    var indexerGroup = ArrayBuilder<PropertySymbol>.GetInstance();
+                    foreach (Symbol symbol in lookupResult.Symbols)
+                    {
+                        Debug.Assert(symbol.IsIndexer());
+                        indexerGroup.Add((PropertySymbol)symbol);
+                    }
+
+                    // PROTOTYPE(list-patterns) Use placeholder
+                    BoundExpression indexerArgumentExpr = new BoundImplicitReceiver(syntax, indexerArgumentType);
+                    var analyzedArguments = AnalyzedArguments.GetInstance(ImmutableArray.Create(indexerArgumentExpr));
+                    var result = BindIndexerOrIndexedPropertyAccess(
+                        syntax, receiverOpt: new BoundImplicitReceiver(syntax, receiverType),
+                        propertyGroup: indexerGroup, analyzedArguments, diagnostics, forListPattern: true);
+                    analyzedArguments.Free();
+                    if (result is BoundIndexerAccess { HasErrors: false } indexerAccess)
```

Comment:
>false [](http://example.com/codeflow?start=66&length=5)

It feels like we should ignore HasErrors here. If it bound to an indexer, it will probably produce better API and IDE results if we just make it the indexer, errors and all.

Reply:
We are not binding the indexer. We're just trying the lookup for a couple of alternative pattern-based members, neither of which should report any errors.

Reply:
Something still feels off here. We're reporting diagnostics here (meaning that we've decided we'll use this member). Shouldn't we report the use-site diagnostics then?

Reply:
Not sure if you've addressed my question: Shouldn't we also report the use-site diagnostics here?

Reply:
See `TryFindIndexerOrIndexerPattern`

This is done a few iterations back. We capture useSiteInfo in the caller when either lookup is succeeded.